From acc0ef6e8cdde58d8a3f5baea6d6066377167c60 Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Fri, 24 Jun 2022 11:22:40 +0530
Subject: [PATCH] ath12k : Add 5GHz 240MHz support

240MHz is supported in 5GHz for single band from channel
100 - 144 with right hand 80MHz puncturing in
320MHz bandwidth.

Updating maximum allowed bandwidth and allow
320MHz operation for the freqencies.

Signed-off-by: Ramya Gnanasekar <quic_rgnanase@quicinc.com>
Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c |  3 ++-
 drivers/net/wireless/ath/ath12k/reg.c | 11 ++++++++---
 2 files changed, 10 insertions(+), 4 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -13367,7 +13367,8 @@ ath12k_mac_setup_radio_iface_comb(struct
 					BIT(NL80211_CHAN_WIDTH_20) |
 					BIT(NL80211_CHAN_WIDTH_40) |
 					BIT(NL80211_CHAN_WIDTH_80) |
-					BIT(NL80211_CHAN_WIDTH_160);
+					BIT(NL80211_CHAN_WIDTH_160) |
+					BIT(NL80211_CHAN_WIDTH_320);
 
 	return 0;
 }
